### ISSUE TITLE
Update builds to JDK16

### ISSUE THREAD

**Issue by lars-sh** [lars-sh/parent#240]
Upgrade build pipelines to JVM 16

**Comment by lars-sh** [comment]
This is currently blocked by:
```
Error:  Failed to execute goal org.projectlombok:lombok-maven-plugin:1.18.18.0:delombok (default) on project utils-test: Execution default of goal org.projectlombok:lombok-maven-plugin:1.18.18.0:delombok failed: An API incompatibility was encountered while executing org.projectlombok:lombok-maven-plugin:1.18.18.0:delombok: java.lang.IllegalAccessError: class lombok.delombok.Delombok (in unnamed module @0x3fd93330) cannot access class com.sun.tools.javac.tree.JCTree$JCCompilationUnit (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.tree to unnamed module @0x3fd93330
Error:  -----------------------------------------------------
Error:  realm =    plugin>org.projectlombok:lombok-maven-plugin:1.18.18.0
Error:  strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
Error:  urls[0] = file:/home/runner/.m2/repository/org/projectlombok/lombok-maven-plugin/1.18.18.0/lombok-maven-plugin-1.18.18.0.jar
Error:  urls[1] = file:/home/runner/.m2/repository/org/projectlombok/lombok/1.18.18/lombok-1.18.18.jar
Error:  urls[2] = file:/home/runner/.m2/repository/org/apache/commons/commons-lang3/3.8.1/commons-lang3-3.8.1.jar
Error:  urls[3] = file:/home/runner/.m2/repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.jar
Error:  urls[4] = file:/home/runner/.m2/repository/org/codehaus/plexus/plexus-utils/1.5.8/plexus-utils-1.5.8.jar
Error:  Number of foreign imports: 1
Error:  import: Entry[import  from realm ClassRealm[project>de.lars-sh:root:0.9.12-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
```

**Comment by lars-sh** [comment]
As of https://projectlombok.org/download-edge this should be fixed when Project Lombok 1.18.20 is released.

**Comment by lars-sh** [comment]
I just tried to apply the given workaround in #248, but as we're using Delombok there's one more step to do, as Delombok will need to fork a new JVM, supplying the compiler args.
